### DISCUSSION TITLE
What sensor type is proper for TCRT5000 in Tasmota?

### DISCUSSION THREAD

**Discussion by cleverpal-ai** [arendst/Tasmota#16113] [Q&A] [answered]
I have an ESP32 with Tasmota and use a TCRT5000 IR distance sensor (connected via GPIO19). 
I do not need any detailed info regarding the distance to the object, just need some discrete figures (0-1, absent-present).
Is it possible to realize with Tasmota? Please advise the most appropriate GPIO selection in " Module parameters".

Thanks!

**Comment by barbudor** [comment, marked as answer]
From what I understand the TCRT5000 falls in the category of IR fence so it should provide a hi/low logic signal that you can catch with a 'Switch' input using 'switchmode 15'

Regarding cabling it should be:
Led part:
3v3 - - - R(150 to 220 ohm) - - - pin A
Pin C - - - GND
Transistor part:
Pin C - - - gpio of your choice except 0, 2 and 15
Pin E - - - GND
You main need to adjust the resistor to adjust sensitivity as far as you don't go lower than 33 ohm

On each change a message will be generated that you can either process with a Rule or through MQTT

**Reply by cleverpal-ai** [reply]
Thanks for your reply!
I cannot find any GPIO type that could resemble an "IR fence". What I'm looking is a proper selection from here:
![image](https://user-images.githubusercontent.com/62514634/181836677-35c291b2-9ca6-4fb0-95a3-08bf5b956d76.png)

As for connection - TCRT5000 works and registers incremental figures if "Counter" is selected in GPIO I'm using. But Counter increments the distance to the object and cannot be used as a trigger.

**Reply by barbudor** [reply]
I wrote a 'switch' input

**Reply by cleverpal-ai** [reply]
@barbudor , Thanks! It works!
My TCRT5000 has only 3 pins (OUT, VCC, GND), and resistor adjustment doesn't let me choose 25-30 cm distance, comfortable for triggering the switch. It triggers only when distance is ~1 cm or 10+ cm. Anyway, it's a bit of off-topic, maybe I will need to change to another sensor.

**Reply by barbudor** [reply]
So you don't have the bare chip but a module on which there should already be a resistor

For distance sensor, look at [VL53Lxx](https://tasmota.github.io/docs/VL53Lxx/) sensors. Depending on the distance range you are looking at should fit
